README.md

```diff
@@ -397,11 +397,94 @@ however `ActivityNew3` will get executed, since the release wasn't yet checked a
 every new execution of the workflow — all new activities will get executed, while `ActivityOld` will
 not.
 
-Later on you can clean it up and drop all the checks if you don't have any older workflows running
-or expect them to ever be executed (e.g. reset).
-
 *NOTE: Releases with different names do not depend on each other in any way.*
 
+## Extras
+
+This section describes optional extra modules included in the SDK for convenience and some
+additional functionality.
+
+### Versioned Workflows
+
+Implemnting breaking changes using the previously described `#has_release?` flag can be error prone
+and results in a condition build up in workflows over time.
+
+Another way of implementing breaking changes is by doing a full cut over to the new version every
+time you need to modify a workflow. This can be achieved manually by treating new versions as
+separate workflows. We've simplified this process by making your workflow aware of its versions:
+
+```ruby
+require 'cadence/concerns/versioned'
+
+class MyWorkflowV1 < Cadence::Workflow
+  retry_policy max_attempts: 5
+
+  def execute
+    Activity2.execute!
+  end
+end
+
+class MyWorkflowV2 < Cadence::Workflow
```

Comment:
Interesting… so if I understand correctly you're doing something very similar — you are picking the latest version at the start of each execution and then adding conditionals to the workflow code to segregate different version. Am I getting it right?

Btw, I don't think you need to memoize the version yourself because `has_release?` does it for you — after it was called ones it will return the same value for all subsequent calls.

We were following the same approach using the `has_release?` explicitly, but found it to be quite noisy.

Maybe the optimal solution is the combination of the two — version is submitted explicitly via headers and then can be checked inline with `version_at_least?`. It allows for feature flag rollouts as well as concise PR diffs.

I like the idea of adding ES support and tagging versions for searchability 👍